cluster/repository/cluster_repository_blackbox_test.go

```diff
@@ -148,43 +161,97 @@ func (s *clusterTestSuite) TestSaveOK() {
 	cluster1.Type = uuid.NewV4().String()
 	cluster1.URL = uuid.NewV4().String()
 	cluster1.CapacityExhausted = true
-
+	// when
 	err := s.repo.Save(context.Background(), cluster1)
+	// then
 	require.NoError(s.T(), err)
-
 	loaded1, err := s.repo.Load(context.Background(), cluster1.ClusterID)
 	require.NoError(s.T(), err)
 	test.AssertEqualClusters(s.T(), cluster1, loaded1)
-
 	loaded2, err := s.repo.Load(context.Background(), cluster2.ClusterID)
 	require.NoError(s.T(), err)
 	test.AssertEqualClusters(s.T(), cluster2, loaded2)
 }
 
 func (s *clusterTestSuite) TestSaveUnknownFails() {
+	// given
 	id := uuid.NewV4()
+	// when
 	err := s.repo.Save(context.Background(), &repository.Cluster{ClusterID: id})
+	// then
 	test.AssertError(s.T(), err, errors.NotFoundError{}, "cluster with id '%s' not found", id)
 }
 
 func (s *clusterTestSuite) TestExists() {
+	// given
 	id := uuid.NewV4()
 	err := s.repo.CheckExists(context.Background(), id.String())
 	test.AssertError(s.T(), err, errors.NotFoundError{}, "cluster with id '%s' not found", id)
-
+	// when
 	cluster := test.CreateCluster(s.T(), s.DB)
+	// then
 	err = s.repo.CheckExists(context.Background(), cluster.ClusterID.String())
 	require.NoError(s.T(), err)
 }
 
 func (s *clusterTestSuite) TestQueryOK() {
+	// given
 	cluster1 := test.CreateCluster(s.T(), s.DB)
-
+	// when
 	clusters, err := s.repo.Query(func(db *gorm.DB) *gorm.DB {
 		return db.Where("cluster_id = ?", cluster1.ClusterID)
 	})
-
+	// then
 	require.NoError(s.T(), err)
 	require.Len(s.T(), clusters, 1)
 	test.AssertEqualClusters(s.T(), cluster1, &clusters[0])
 }
+
+func (s *clusterTestSuite) TestList() {
+	// given
+	cluster1 := &repository.Cluster{
```

Comment:
You can just use https://github.com/fabric8-services/fabric8-cluster/blob/master/test/cluster.go#L18 to create a random cluster in DB.

Reply:
cool, thanks!

Reply:
done in dc31c0f